2.0: DatabaseTest: any tests with xxDifferentDBInstance fail

- [x] testGetExistingDocWithIDFromDifferentDBInstance
- [x] testSaveDocInDifferentDBInstance
- [x] testDeleteDocInDifferentDBInstance
- [x] testPurgeDocInDifferentDBInstance

```
CouchbaseLiteException{domain=LiteCore, code=24}
at com.couchbase.lite.internal.bridge.LiteCoreBridge.convertException(LiteCoreBridge.java:21)
at com.couchbase.lite.Database.delete(Database.java:383)
at com.couchbase.lite.BaseTest.tearDown(BaseTest.java:73)
at com.couchbase.lite.DatabaseTest.tearDown(DatabaseTest.java:148)
at java.lang.reflect.Method.invoke(Native Method)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:33)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at android.support.test.internal.runner.TestExecutor.execute(TestExecutor.java:59)
at android.support.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:262)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:1879)
Caused by: com.couchbase.litecore.LiteCoreException: Still 1 open connection(s) to /data/user/0/com.couchbase.lite.test/files/CouchbaseLite/testdb.cblite2/db.sqlite3
at com.couchbase.litecore.Database.deleteAtPath(Native Method)
at com.couchbase.lite.Database.delete(Database.java:381)
... 29 more
```

--- a/android/CouchbaseLite/src/androidTest/java/com/couchbase/lite/DatabaseTest.java
+++ b/android/CouchbaseLite/src/androidTest/java/com/couchbase/lite/DatabaseTest.java
@@ -50,12 +50,16 @@ public class DatabaseTest extends BaseTest {
 
     // helper method to open database
     private Database openDatabase(String dbName) throws CouchbaseLiteException {
+        return openDatabase(dbName, true);
+    }
+    private Database openDatabase(String dbName, boolean countCheck) throws CouchbaseLiteException {
         DatabaseConfiguration options = new DatabaseConfiguration(this.context);
         options.setDirectory(dir);
         Database db = new Database(dbName, options);
         assertEquals(dbName, db.getName());
         assertTrue(db.getPath().getAbsolutePath().endsWith(".cblite2"));
-        assertEquals(0, db.getCount());
+        if(countCheck)
+            assertEquals(0, db.getCount());
         return db;
     }
 
@@ -334,14 +338,14 @@ public class DatabaseTest extends BaseTest {
         verifyGetDocument(docID);
     }
 
-    // TODO: @Test
+    @Test
     public void testGetExistingDocWithIDFromDifferentDBInstance() throws CouchbaseLiteException {
         // store doc
         String docID = "doc1";
         generateDocument(docID);
 
         // open db with same db name and default option
-        Database otherDB = openDatabase(db.getName());
+        Database otherDB = openDatabase(db.getName(), false);
         assertNotNull(otherDB);
         assertTrue(db != otherDB);
 
@@ -439,15 +443,14 @@ public class DatabaseTest extends BaseTest {
         verifyGetDocument(docID, 2);
     }
 
-    // TODO : @Test
-    // LiteCore throws exception in tearDown
+    @Test
     public void testSaveDocInDifferentDBInstance() throws CouchbaseLiteException {
         // Store doc
         String docID = "doc1";
         Document doc = generateDocument(docID);
 
         // Create db with default
-        Database otherDB = openDatabase(db.getName());
+        Database otherDB = openDatabase(db.getName(), false);
         assertNotNull(otherDB);
         assertTrue(otherDB != db);
         assertEquals(1, otherDB.getCount());
@@ -580,8 +583,7 @@ public class DatabaseTest extends BaseTest {
         assertTrue(doc.getObject("key") == null);
     }
 
-    // TODO : @Test
-    // Error from LiteCore in tearDown
+    @Test
     public void testDeleteDocInDifferentDBInstance() throws CouchbaseLiteException {
         // Store doc:
         String docID = "doc1";
@@ -589,7 +591,7 @@ public class DatabaseTest extends BaseTest {
 
         // Create db with same name:
         // Create db with default
-        Database otherDB = openDatabase(db.getName());
+        Database otherDB = openDatabase(db.getName(), false);
         assertNotNull(otherDB);
         assertTrue(otherDB != db);
         assertTrue(otherDB.contains(docID));
@@ -744,15 +746,14 @@ public class DatabaseTest extends BaseTest {
         assertEquals(2, doc.getSequence());
     }
 
-    // TODO : @Test
-    // Error from LiteCore in tearDown
+    @Test
     public void testPurgeDocInDifferentDBInstance() throws CouchbaseLiteException {
         // Store doc:
         String docID = "doc1";
         Document doc = generateDocument(docID);
 
         // Create db with default:
-        Database otherDB = openDatabase(db.getName());
+        Database otherDB = openDatabase(db.getName(), false);
         assertNotNull(otherDB);
         assertTrue(otherDB != db);
         assertEquals(1, otherDB.getCount());
@@ -1171,7 +1172,7 @@ public class DatabaseTest extends BaseTest {
         assertFalse(Database.exists("nonexist", dir));
     }
 
-    //TODO: @Test - test fails because database.compact() does not delete blob files.
+    @Test
     public void testCompact() throws CouchbaseLiteException {
         final List<Document> docs = createDocs(20);
 

